src/re-rxjs/index.js

```diff
@@ -0,0 +1,79 @@
+import React, { unstable_useTransition as useTransition } from 'react';
+import { Subject, merge, asapScheduler } from 'rxjs';
+import {
+  map, scan, startWith, observeOn,
+} from 'rxjs/operators';
+import { connectObservable } from 're-rxjs';
+import {
+  syncBlock,
+  useRegisterIncrementDispatcher,
+  ids,
+  useCheckTearing,
+} from '../common';
+
+const normalClicks$ = new Subject();
+const normalIncrement = () => normalClicks$.next();
+
+const externalClicks$ = new Subject();
+// This is only needed to prevent the tearing with auto increment (check #9)
```

Comment:
check 9 is to distinguish useSubscription and useMutableSource, so you wouldn't need to treat it specially. I don't know how `asapScheduler` helps here, but the purpose of the check is not to bypass like that. (I mean, that means the check 9 is not working as I expect.)

Reply:
Both `useSubscription` and `useMutableSource` are custom hooks built on top of the existing primitive hooks, meaning that they don't do anything that can't be done from user-land.

I've learned a lot from understanding what they do, and in early versions I was using `useSubscription`. However, once I better understood what the hook was doing and how RxJS subscriptions behave, I decided to build my own integration for RxJS subscriptions.

The `asapScheduler` doesn't try to "bypass" a test. That's not at all what it is for. I will document this, but the thing is that when working with the RxJS bindings, if you are subscribed to a data-source that comes from outside of React, then you should be observing it with the `asapScheduler` because otherwise react won't be able to batch the updates.

So, no, I'm not trying to cheat here :slightly_smiling_face: this is how the bindings should be used with RxJS for inputs that don't come from withing React event-handlers.